src/Controller/CreateCaseController.php

```diff
@@ -58,6 +60,17 @@ public function createCase(BoardRepository $boardRepository, MunicipalityReposit
         if ($form->isSubmitted() && $form->isValid()) {
             $case = $form->getData();
 
+            $lock = $lockFactory->createLock('case-number-generation');
+
+            $lock->acquire(true);
+
+            // case number generation
+            $caseNumber = $caseManager->generateCaseNumber($municipality);
+
+            $lock->release();
```

Comment:
You should probably release the lock after the new case is persisted in the database, otherwise it's still possible for another process to generate the same case-number... Unlikely though but still.

Reply:
Good catch.  Release of lock now happens after flushing.